In [1]:
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt
%matplotlib inline

In [4]:
pd.read_csv(r"C:\Users\User\Documents\Datasets\test_lAUu6dG.csv")
loan = pd.read_csv(r"C:\Users\User\Documents\Datasets\test_lAUu6dG.csv")

In [9]:
loan.isnull().sum()

Loan_ID               0
Gender               11
Married               0
Dependents           10
Education             0
Self_Employed        23
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount            5
Loan_Amount_Term      6
Credit_History       29
Property_Area         0
dtype: int64

In [22]:
loan[loan.Credit_History == 0]

Loan_ID  Gender Married Dependents     Education Self_Employed  \
7    LP001056    Male     Yes          2  Not Graduate            No   
13   LP001094    Male     Yes          2      Graduate           NaN   
25   LP001153    Male      No          0      Graduate            No   
35   LP001203    Male      No          0      Graduate            No   
55   LP001313    Male      No          0      Graduate            No   
58   LP001323  Female     Yes          2      Graduate            No   
63   LP001347  Female      No          0      Graduate            No   
66   LP001352    Male     Yes          0  Not Graduate            No   
67   LP001358    Male     Yes          0      Graduate            No   
69   LP001361    Male     Yes          0      Graduate            No   
80   LP001420    Male     Yes         3+      Graduate            No   
82   LP001445    Male     Yes          2  Not Graduate            No   
84   LP001450    Male     Yes          0      Graduate            No   
94   LP001496    Male     Yes          1      Graduate            No   
101  LP001542  Female     Yes          0      Graduate            No   
106  LP001563     NaN      No          0      Graduate            No   
117  LP001611    Male     Yes          1      Graduate            No   
118  LP001613  Female      No          0      Graduate            No   
119  LP001622    Male     Yes          2      Graduate            No   
123  LP001652    Male      No          0      Graduate            No   
124  LP001655  Female      No          0      Graduate            No   
126  LP001662    Male      No          0      Graduate            No   
140  LP001785    Male      No          0      Graduate            No   
142  LP001789    Male     Yes         3+  Not Graduate           NaN   
147  LP001817    Male      No          0  Not Graduate           Yes   
153  LP001853    Male     Yes          1  Not Graduate            No   
161  LP001906    Male      No          0      Graduate           NaN   
165  LP001923    Male      No          0  Not Graduate            No   
166  LP001933    Male      No          2  Not Graduate            No   
168  LP001950  Female     Yes         3+      Graduate           NaN   
173  LP001979    Male      No          0      Graduate            No   
174  LP001995    Male     Yes          0  Not Graduate            No   
192  LP002069    Male     Yes          2  Not Graduate           NaN   
193  LP002070    Male     Yes         3+  Not Graduate            No   
196  LP002090    Male     Yes          1      Graduate            No   
198  LP002099    Male     Yes          2      Graduate            No   
211  LP002168    Male      No          0      Graduate            No   
224  LP002256  Female      No          2      Graduate           Yes   
229  LP002286    Male     Yes         3+  Not Graduate            No   
235  LP002316    Male      No          0      Graduate            No   
236  LP002321  Female      No          0      Graduate            No   
241  LP002339    Male     Yes          0      Graduate            No   
243  LP002346    Male     Yes          0      Graduate           NaN   
245  LP002355     NaN     Yes          0      Graduate            No   
250  LP002383    Male     Yes         3+      Graduate            No   
255  LP002399    Male      No          0      Graduate           NaN   
266  LP002442  Female     Yes          1  Not Graduate            No   
268  LP002450    Male     Yes          2      Graduate            No   
273  LP002495    Male     Yes          2      Graduate            No   
274  LP002496  Female      No          0      Graduate            No   
278  LP002551    Male     Yes         3+  Not Graduate           NaN   
293  LP002609  Female     Yes          0      Graduate            No   
301  LP002651    Male     Yes          1      Graduate           NaN   
311  LP002747  Female      No         3+      Graduate            No   
317  LP002774    Male     Yes          0  Not 

In [24]:
loan.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       367.000000         367.000000  362.000000        361.000000   
mean       4805.599455        1569.577657  136.132597        342.537396   
std        4910.685399        2334.232099   61.366652         65.156643   
min           0.000000           0.000000   28.000000          6.000000   
25%        2864.000000           0.000000  100.250000        360.000000   
50%        3786.000000        1025.000000  125.000000        360.000000   
75%        5060.000000        2430.500000  158.000000        360.000000   
max       72529.000000       24000.000000  550.000000        480.000000   

       Credit_History  
count      338.000000  
mean         0.825444  
std          0.380150  
min          0.000000  
25%          1.000000  
50%          1.000000  
75%          1.000000  
max          1.000000

In [5]:
loan.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            367 non-null    object 
 1   Gender             356 non-null    object 
 2   Married            367 non-null    object 
 3   Dependents         357 non-null    object 
 4   Education          367 non-null    object 
 5   Self_Employed      344 non-null    object 
 6   ApplicantIncome    367 non-null    int64  
 7   CoapplicantIncome  367 non-null    int64  
 8   LoanAmount         362 non-null    float64
 9   Loan_Amount_Term   361 non-null    float64
 10  Credit_History     338 non-null    float64
 11  Property_Area      367 non-null    object 
dtypes: float64(3), int64(2), object(7)
memory usage: 34.5+ KB


The dataset contains 367 entries for loan applications. The features includes Loan_ID, Gender, Marrital status, education and other loan details. There are 11 null values under the gender column, 10 null values under dependents, 23 null values under self-employed, 5 null values under loan amounts, 6 under loan amount term and 29 under credit history. 
The aim of my exploration is to answer some questions

The relationship between income, nature of employment and loan amount
Credit history and 
frequency of unemployed rural dwellers taking loans
relationship between education and credit history, loan amount.
The relationship between Loan amount and marital status
The percentage of married men who take loans
Comparison between married woman and single women

In [23]:
loan.head(3)

Loan_ID Gender Married Dependents Education Self_Employed  \
0  LP001015   Male     Yes          0  Graduate            No   
1  LP001022   Male     Yes          1  Graduate            No   
2  LP001031   Male     Yes          2  Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5720                  0       110.0             360.0   
1             3076               1500       126.0             360.0   
2             5000               1800       208.0             360.0   

   Credit_History Property_Area  
0             1.0         Urban  
1             1.0         Urban  
2             1.0         Urban

Quality issues
- Adress Null values
- Applicant Income descripancies
- Convert Loan amount and Amount term to integers
- 
Tidyness issues
- Loan ID as Primary Key
- 